### ISSUE TITLE
Unable to create MX records with "." as value

### ISSUE THREAD

**Issue by AlessioCasco** [cloudflare/terraform-provider-cloudflare#944]
Hi there,

I'm unable to create an MX record with `.` as value.
This is for protecting a parked domain and setting a null MX

### Terraform version

Terraform v0.14.6
+ provider registry.terraform.io/cloudflare/cloudflare v2.18.0

### Affected resource(s)

- cloudflare_record

### Debug output
https://gist.github.com/AlessioCasco/8adb27147ecf21b6c4e809ab108f9294

### Expected behavior

The MX record should be created with "." as value.
I am able to do it using the Cloudflare website

### Actual behavior
Terraform exits with an error:
```bash
2021/02/09 18:15:39 [DEBUG] module.cloudflare_zone.cloudflare_record.mx: apply errored, but we're indicating that via the Error pointer rather than returning it: failed to create DNS record: error from makeRequest: HTTP status 400: content "{\"result\":null,\"success\":false,\"errors\":[{\"code\":1004,\"message\":\"DNS Validation Error\",\"error_chain\":[{\"code\":9009,\"message\":\"Content for MX record must be a hostname.\"}]}],\"messages\":[]}"
```

### Steps to reproduce

1. `terraform apply`

### References
The code above has priority 1 due to the following issue:
- GH-943

### Community note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)
  to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull
  request, please leave a comment

**Comment by jacobbednarz** [comment]
Just looking into this one as I'm not sure if you should be able to use `.` as a value (in either the UI or API). The RFC for DNS MX records denotes the value as needing to be a full domain name, not just any character.

**Comment by tomelfring** [comment]
Using the value "." with priority one would be conform https://tools.ietf.org/html/rfc7505

**Comment by jacobbednarz** [comment]
thanks for the link @tomelfring, however, it looks like rfc7505 is still only a proposed standard and i'm unsure if the UI is intentional support or just works because of an edge case. let me confirm to make sure we're not opening a can of worms here we shouldn't be.

**Comment by jacobbednarz** [comment]
confirmed the support is indeed intentional! let's line this up to be included.

**Comment by jacobbednarz** [comment]
apologies, i completed overlooked the actual cause of this and it's the fact you're sending a priority of 1 with the content of "." (which is invalid) and the API itself it rejecting it, not Terraform. this will be possible once #943 lands so no action required here.
